app/src/main/java/org/schabi/newpipe/info_list/dialog/StreamDialogDefaultEntry.java

```diff
@@ -110,6 +116,35 @@ public enum StreamDialogDefaultEntry {
             ShareUtils.shareText(fragment.requireContext(), item.getName(), item.getUrl(),
                     item.getThumbnailUrl())),
 
+    DOWNLOAD(R.string.download, (fragment, item) -> ExtractorHelper.getStreamInfo(
+                    item.getServiceId(),
+                    item.getUrl(),
+                    false
+            )
+            .subscribeOn(Schedulers.io())
```

Comment:
Shouldn't you `observeOn(AndroidSchedulers.mainThread())` to make sure the code in the `.subscribe()` is run on the UI thread?

Reply:
Resolved